pkg/deploy/api/validation/validation.go

```diff
@@ -39,6 +39,39 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	if spec.Template == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
 	} else {
+		// if we have both an ICT defined and an empty Template->PodSpec->Container->Image field, we are going
+		// to modify this method's local copy (a pointer was NOT used for the parameter) by setting the field to a non-empty value to
+		// work around the k8s validation as our ICT will supply the image field value
+		containerEmptyImageInICT := make(map[string]bool)
+		for _, container := range spec.Template.Spec.Containers {
+			if len(container.Image) == 0 {
+				containerEmptyImageInICT[container.Name] = false
+			}
+		}
+		needToChangeImageField := false
+		if len(containerEmptyImageInICT) > 0 {
+			for _, trigger := range spec.Triggers {
+				if trigger.Type == deployapi.DeploymentTriggerOnImageChange {
+					// note, the validateTrigger call above will add an error if ImageChangeParams is nil, but
+					// we can still fall down this path so account for it being nil
+					if trigger.ImageChangeParams != nil {
+						for _, container := range trigger.ImageChangeParams.ContainerNames {
+							needToChangeImageField = true
+							containerEmptyImageInICT[container] = true
+						}
+					}
+				}
+			}
+		}
+		if needToChangeImageField {
+			for i, container := range spec.Template.Spec.Containers {
+				// only update containers listed in the icp
+				match, ok := containerEmptyImageInICT[container.Name]
```

Comment:
it seems like you can get rid of this whole structure.  all you need to do in this loop is again check if the container.Image field is length 0, and if so, update it.